src/avram/queryable.cr

```diff
@@ -122,6 +122,14 @@ module Avram::Queryable(T)
     clone.tap &.query.where(sql_clause)
   end
 
+  def where : self
+    query.where(Avram::Where::PrecedenceStart.new)
```

Comment:
I _think_ you're going to need a clone call before this. If you assign the query to the variable before calling this where without the clone it will have the opening parenthesis but nothing else on it after.

Reply:
oh, interesting. Ok, I'll add a spec for that. Thanks

Reply:
Good catch. What it does is adds a blank `WHERE`. `SELECT * FROM users WHERE ;`... But after adding the clone, now I have a new issue where it won't add the `(` to the query. `SELECT * FROM users WHERE age = $1 );`

```crystal
clone.tap &.query.where(Avram::Where::PrecedenceStart.new)
result = yield self
result.query.clear_conjunction
result.query.where(Avram::Where::PrecedenceEnd.new)
result
```

If I throw the whole thing inside of the `clone.tap`, then it adds the parens, but omits the actual where clauses.

Reply:
Ahh... Ok, that works. But wow.... All this cloning has to be expensive, right? 🤔